python/src/phylanx.cpp

```diff
@@ -497,36 +535,390 @@ PYBIND11_MODULE(_phylanx, m)
+            blaze::DynamicMatrix<double> m{nx, ny};
+            m = 0.0;
+            return hpx::threads::run_as_hpx_thread([&]() {
+                using namespace phylanx::execution_tree;
+                return primitive{hpx::local_new<primitives::variable>(
+                    phylanx::ir::node_data<double>{m})};
+            });
+        },
+        "create a new variable from a matrix of zeros");
+
+    execution_tree.def(
+        "phylisp_eval", expression_compiler<>, expression_compiler_help);
+    execution_tree.def(
+        "phylisp_eval", expression_compiler<double>, expression_compiler_help);
+    execution_tree.def("phylisp_eval",
+        expression_compiler<phylanx::execution_tree::primitive const&>,
+        expression_compiler_help);
+    execution_tree.def("phylisp_eval",
+        expression_compiler<phylanx::execution_tree::primitive const&,
+            phylanx::execution_tree::primitive const&>,
+        expression_compiler_help);
+    execution_tree.def("phylisp_eval",
+        expression_compiler<phylanx::execution_tree::primitive const&, double>,
+        expression_compiler_help);
+    execution_tree.def("phylisp_eval",
+        expression_compiler<double, phylanx::execution_tree::primitive const&>,
+        expression_compiler_help);
+    execution_tree.def("phylisp_eval",
+        expression_compiler<double, double>,
+        expression_compiler_help);
+    execution_tree.def("phylisp_eval",
+        expression_compiler<phylanx::execution_tree::primitive const&,
+            phylanx::execution_tree::primitive const&,
+            phylanx::execution_tree::primitive const&>,
+        expression_compiler_help);
+    execution_tree.def("phylisp_eval",
+        expression_compiler<double, phylanx::execution_tree::primitive const&,
+            phylanx::execution_tree::primitive const&>,
+        expression_compiler_help);
+    execution_tree.def("phylisp_eval",
+        expression_compiler<phylanx::execution_tree::primitive const&, double,
+            phylanx::execution_tree::primitive const&>,
+        expression_compiler_help);
+    execution_tree.def("phylisp_eval",
+        expression_compiler<phylanx::execution_tree::primitive const&,
+            phylanx::execution_tree::primitive const&, double>,
+        expression_compiler_help);
+    execution_tree.def("phylisp_eval",
+        expression_compiler<double, double,
+            phylanx::execution_tree::primitive const&>,
+        expression_compiler_help);
+    execution_tree.def("phylisp_eval",
+        expression_compiler<double, phylanx::execution_tree::primitive const&,
+            double>,
+        expression_compiler_help);
+    execution_tree.def("phylisp_eval",
+        expression_compiler<phylanx::execution_tree::primitive const&, double,
+            double>,
+        expression_compiler_help);
+    execution_tree.def("phylisp_eval",
+        expression_compiler<double, double, double>,
+        expression_compiler_help);
```

Comment:
Wouldn't it be sufficient defining overload of this function taking one or more instances of `primitive_argument_type`'s? Otherwise we'll get an exponential explosion of overloads for all possible combination of argument types.

Reply:
I don't see a way to do the autoconversion and it's a pain to do et.phylisp_eval("3") instead of just 3. Not sure what the right answer is here.

Reply:
Rereading your comment... So if I use primitive_argument_type, I need a way to create a python object of that type. Currently I don't have one and Python doesn't autoconvert double or anything else to that type.

Reply:
`primitive_argument_type` should be constructable from `double` - so I assumed the binding would figure it out. Let me experiment a bit before getting back to this.